python_modules/dagster/dagster_tests/core_tests/execution_tests/test_context.py

```diff
@@ -20,3 +23,24 @@ def foo():
         ctx_op()
 
     assert foo.execute_in_process().success
+
+
+def test_instance_check():
+    # turn off any outer warnings filters, e.g. ignores that are set in pyproject.toml
+    warnings.resetwarnings()
+    warnings.filterwarnings("error")
+
+    @op
+    def test_op_context_instance_check(context: OpExecutionContext):
+        step_context = context._step_execution_context  # noqa: SLF001
```

Comment:
Alex: what about @op context: OpExecutionContext doing this isinstance check ?

[source](https://github.com/dagster-io/dagster/pull/16762#discussion_r1336233357)

Reply:
I think this version of the test takes care of that concern, since we're just using the dagster provided context to get a `StepExecutionContext` we use to create our own `AssetExecutionContext` and `OpExecutionContext`